### PR TITLE
refactor: split icons to packages

### LINKED ISSUE #240: Wrangler Deploy Fails After Adding FrogUI
### Describe the bug

Receiving Cloudflare error `workers.api.error.script_too_large [code: 10027]` after adding FrogUI

### Link to Minimal Reproducible Example

[TypeScript Playground](https://www.typescriptlang.org/play?#code/JYWwDg9gTgLgBAbzgIQK4xhAdgGjgMSggHM4BfOAMyJDgHJqS6BuAKFElkRQgA88AEgFMAhgBNgWYngBqAZRgiAxgGs8ANxFQAzuSo16AOgD0qYIYBW2lu3DR4SMUPWYIAG10VGtBkWLGndRsOe25tISh1IQURGGAlPW96Rn9wyKEAWm1FOKUbY2M4EK4kLCEATywtRINfEmMAC1QAI2tWViFeTnglbGy4ETAwOABeODKAdwI-AAoEVjg4ZqIJtIAZCCVY4GwALnpjOhwFuAK4OVQht3KBuAEWuEw4ISrmtyF9ERAPqKhgSni22whhOZyazX2ZUqWjmAzAwAA0hV9nQAHIAUQAmqiAIIAJQA+vg8QB5ADiRIAsnRyABKE5mfZITQ6cjHMi0wzUL5CGZ0Q54BIjAB8iBOUCEMFQUCwcCUhgl2jmJ0WyjiewObhI2COKqKIBExCE+xmesWAB5kHwzYs4MQVjbVW5gMQsAJoMAAF7YRRuEYAIiULxgEX9joGztdMgiuREfsDwdD4eayhU9ogqCwYgAwu5oAGU6p05mxGHbeWltAnFBc1qoAGJg1gCGyxXFs0qxEAOrAMQwBoBgBsrbbYHEEikAYAzAAmEe24Xh83yRSqRdt8vm4TiSSkbReoRDgAs-uFcgacbccC7HwACm4RDcRNMeQB+c3GbcT4jrjcflepr+m7GFavBAXAtLHOWkghlgMDaPsADalroJgsqaG4qCHv6YAPuUp73o+H5oBg2DCgAulBdJsBybCsL0WD9MA2jZlqqBiJQD4Sl20AqBEoyPOUYBCBAlBysoDRCLoACEIxjHQJZCACZRiMEYkzMxrEZhxXFCDxUB8VAtJiosDH9GkUQxLkJJgOqjECUgRAQDAKImEccAGlg-xSS5iAUGQbCqkMhioOEfLGAAVO5FnRDk8QzDFVnxDZdnaLStKBXAgSuB4MyDGAeBICI2jhPBt6xA0KLGCk0URJZcVKHgiUNSlOz2bRrAUEIHgfPMizZRA7hKvlhVwM12wJB1ZDtJ03RZUpIioG48D5WwQA)

### Steps To Reproduce

Trying to get 🐸 working with [Cloudflare Workers](https://frog.fm/platforms/cloudflare-workers)

```bash
pnpm create frog -t cloudflare-worker
```

In `index.tsx` changed:

```tsx
const manifest = await import('__STATIC_CONTENT_MANIFEST')
const serveStaticOptions = { manifest, root: './' }
```

to:

```tsx
const serveStaticOptions = { root: './', manifest: {} };
```

Added the top level `main` field to `workers.toml`:

```toml
main = "workers-site/index.js"
```

```bash
pnpm wrangler deploy --minify src/index.tsx
```

And it successfully deployed:

```bash
Total Upload: 2334.26 KiB / gzip: 831.76 KiB
```

Then I added `ui.tsx`, imported it, and added some styles in `index.tsx`

```tsx
import { Box, Heading, VStack, vars } from './ui.js';
```

Attempted to deploy the updates:

```bash
pnpm wrangler deploy --minify src/index.tsx
```

And it is failing with the following:

```bash
▲ [WARNING] We recommend keeping your script less than 1MiB (1024 KiB) after gzip. Exceeding this can affect cold start time. Consider using Wrangler's `--minify` option to reduce your bundle size.

▲ [WARNING] Here are the 5 largest dependencies included in your script:

- node_modules/.pnpm/frog@0.8.1_hono@4.2.3/node_modules/frog/_lib/ui/icons.js - 1841.18 KiB
- node_modules/.pnpm/workers-og@0.0.24/node_modules/workers-og/dist/resvg-LFIOYO65.wasm - 1341.69 KiB
- node_modules/.pnpm/workers-og@0.0.24/node_modules/workers-og/dist/index.js - 400.18 KiB
- node_modules/.pnpm/workers-og@0.0.24/node_modules/workers-og/dist/yoga-ZMNYPE6Z.wasm - 86.58 KiB
- node_modules/.pnpm/@scure+bip39@1.2.1/node_modules/@scure/bip39/esm/wordlists/korean.js - 71.91 KiB

If these are unnecessary, consider removing them

✘ [ERROR] A request to the Cloudflare API (/accounts/4e1a03d9fee3ff0defb22eb9feb2c42e/workers/scripts/frog-template) failed.

workers.api.error.script_too_large [code: 10027]
```

```bash
Total Upload: 4185.50 KiB / gzip: 1082.08 KiB
```

### Frog Version

0.8.1

### TypeScript Version

5.4.5

### Check existing issues

- [X] I checked there isn't [already an issue](https://github.com/wevm/frog/issues) for the bug I encountered.

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by dalechyn**
Should fix #240 by decreasing the bundle size by splitting icons in multiple proxy packages.

As of now, all icons are imported even if just one set is used.

### CODE REVIEW

File: src/package.json [hunk 1]
```diff
@@ -105,10 +117,7 @@
   "license": "MIT",
   "homepage": "https://frog.fm",
   "repository": "wevm/frog",
-  "authors": [
-    "awkweb.eth",
-    "jxom.eth"
-  ],
+  "authors": ["awkweb.eth", "jxom.eth"],
```
Comment (dalechyn) [inline src/package.json:124]:
```suggestion
  "authors": [
    "awkweb.eth",
    "jxom.eth"
  ],
```

File: src/ui/Icon.tsx [hunk 2]
```diff
@@ -22,13 +22,11 @@ export type IconProps<
   /**
    * Icon collection to use for resolving icons.
    *
-   * @default 'lucide'
+   * @default lucide from 'frog/ui/lucide'
```
Comment (tmm) [inline src/ui/Icon.tsx]:
```suggestion
   * @default lucide (from 'frog/ui/icons')
```

File: src/package.json
Comment (tmm) [inline src/package.json:1]: Can we add a `'frog/ui/icons'` entrypoint that exports all the icon sets for folks that aren't in a constrained environment? Makes it easier to swap collections and try things out (also looks nicer).

```ts
import { lucide, heroicons } from 'frog/ui/icons'
```

File: site/pages/ui/ui-system.mdx [hunk 3]
```diff
@@ -270,9 +270,10 @@ The `icons` variable is used to set the icon collection for the [`<Icon>` compon
 /** @jsxImportSource frog/jsx */
 // ---cut---
 import { createSystem } from 'frog/ui'
+import { lucide } from 'frog/ui/icons/lucide'
```
Comment (tmm) [inline site/pages/ui/ui-system.mdx]:
```suggestion
import { lucide } from 'frog/ui/icons'
```

File: site/pages/ui/createSystem.mdx [hunk 4]
```diff
@@ -170,9 +170,10 @@ Icon collection to use for resolving icons. The following collections are availa
 
 ```tsx twoslash
 import { createSystem } from 'frog/ui'
+import { lucide } from 'frog/ui/icons/lucide' // [!code focus]
```
Comment (tmm) [inline site/pages/ui/createSystem.mdx]:
```suggestion
import { lucide } from 'frog/ui/icons' // [!code focus]
```

File: site/pages/ui/createSystem.mdx [hunk 5]
```diff
@@ -159,8 +159,8 @@ export const system = createSystem({
 
 ### icons
 
-- **Type:** `'heroicons' | 'lucide' | 'radix-icons'`
-- **Default:** `'lucide'`
+- **Type:** `Record<string, string>`
+- **Default:** `frog/ui/icons/lucide`
```
Comment (tmm) [inline site/pages/ui/createSystem.mdx]:
```suggestion
- **Default:** lucide
```

File: site/pages/ui/Icon.mdx [hunk 6]
```diff
@@ -116,19 +116,20 @@ function Example() {
...
+- [Lucide - `frog/ui/icons/lucide`](https://lucide.dev)
+- [Radix Icons - `frog/ui/icons/radix-icons`](https://www.radix-ui.com/icons)
 
 Collection is mapped to the [`icons` property](/ui/ui-system#icons) on the UI System Variables.
 
 :::code-group
 ```tsx twoslash [Code]
 /** @jsxImportSource frog/jsx */
 import { createSystem } from 'frog/ui'
+import { heroicons } from 'frog/ui/icons/heroicons' // [!code focus]
```
Comment (tmm) [inline site/pages/ui/Icon.mdx]:
```suggestion
import { heroicons } from 'frog/ui/icons' // [!code focus]
```

File: site/pages/ui/Icon.mdx [hunk 7]
```diff
@@ -116,19 +116,20 @@ function Example() {
 
 ### `collection`
 
-Icon collection to use for resolving icons. Defaults to `'lucide'`.
+Icon collection to use for resolving icons. Defaults to `lucide` imported from `'frog/ui/icons/lucide'`.
```
Comment (tmm) [inline site/pages/ui/Icon.mdx]:
```suggestion
Icon collection to use for resolving icons. Defaults to `lucide` imported from `'frog/ui/icons'`.
```

File: playground/src/ui-system.tsx [hunk 8]
```diff
@@ -1,5 +1,8 @@
 import { Button, Frog } from 'frog'
 import { serveStatic } from 'frog/serve-static'
+import { lucide } from 'frog/ui/icons/lucide'
+import { radixIcons } from 'frog/ui/icons/radix-icons'
+import { heroicons } from '../../src/_lib/ui/icons/heroicons/index.js'
```
Comment (tmm) [inline playground/src/ui-system.tsx]:
```suggestion
import { heroicons, lucide, radixIcons } from 'frog/ui/icons'
```

File: .changeset/many-schools-reflect.md [hunk 9]
```diff
@@ -0,0 +1,11 @@
...
+"frog": minor
+---
+
+Split icons into different packages and reduces the bundle size by importing just one collection instead of all.
+Introduces a breaking change as afterwards icons must be imported as packages, rather than literal string.
+Specifically, the next subpackages were created:
+
+- `frog/ui/icons/heroicons`
+- `frog/ui/icons/lucide`
+- `frog/ui/icons/radix-icons`
```
Comment (tmm) [inline .changeset/many-schools-reflect.md]:
```suggestion
**Breaking change** Frog UI `icon` property requires an icon map imported from the `'frog/ui/icons'` entrypoint. This also makes it easier for you to supply your own custom icons.

\```diff
+ import { lucide } from 'frog/ui/icons'

export const system = createSystem({
- icons: 'lucide',
+ icons: lucide,
})
\```

In addition, the following separate entrypoints were added for resource constrained environments.

- `frog/ui/icons/heroicons`
- `frog/ui/icons/lucide`
- `frog/ui/icons/radix-icons`
```

**Review by tmm — CHANGES_REQUESTED** [review]

File: src/package.json
Comment (dalechyn) [reply src/package.json:1]: a47a79a

**Review by tmm — APPROVED** [review]

**Pull request merged**